### PR TITLE
Rewrote /dbdump to Go

### LINKED ISSUE #3785: Rewrite /dbdump to Go
## I'm submitting an ...
improvement request (usability, performance, tech debt, etc.)

## Traffic Control components affected ...
- Documentation (possibly)
- Traffic Ops

## Current behavior:
`/dbdump` is handled by Perl

## Expected / new behavior:
`/dbdump` should be handled by Go.

### Minimum API Version
1.1

### Request Methods
- [ ] `GET`

## Minimal reproduction of the problem with instructions:
N/A

### PR DESCRIPTION
**Pull request by ocket8888**
## What does this PR (Pull Request) do?

- [x] This PR fixes #3785 

## Which Traffic Control components are affected by this PR?

- Documentation
- Traffic Control Client 
- Traffic Ops

## What is the best way to verify this PR?
Call the endpoint and use the response data as the input to `pg_restore` - verify the resulting database is identical to the one dumped at the time it was dumped. I don't know how that could be tested as a unit test, so I didn't write one. I also think it's generally inadvisable to use this endpoint, as a full dump of what is essentially an implementation detail of the API through the API itself seems like an un-document-able nightmare. I'd advise system engineers in need of recreating and/or backing up the database do so by directly using the `pg_dump` tool, rather than the TO API.

## The following criteria are ALL met by this PR
- [x] I have explained why tests are unnecessary
- [x] This PR includes documentation
- [x] This PR includes an update to CHANGELOG.md
- [x] This PR includes any and all required license headers
- [x] This PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY**

### CODE REVIEW

File: traffic_ops/traffic_ops_golang/dbdump/dbdump.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+		userErr = errors.New("'pg_dump' not available")
+		errCode = http.StatusServiceUnavailable
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+
+	cmd := exec.Cmd{
+		Path: pgdump,
+		Args: []string{
+			"--blobs",
```
[the remark this answers is not in the document]
Comment (ocket8888) [reply traffic_ops/traffic_ops_golang/dbdump/dbdump.go]: Yeah, I started a mailing list discussion about the format change

[the remark this answers is not in the document]
Comment (ocket8888) [reply traffic_ops/traffic_ops_golang/dbdump/dbdump.go]: Mailing list decision was to maintain the custom format - which I've done now.

File: infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile [hunk 2]
```diff
@@ -36,7 +36,7 @@ RUN go get -u github.com/jstemmer/go-junit-report
 RUN cd /go/src/github.com/jstemmer/go-junit-report && go build
 
 FROM debian:stretch
-RUN apt-get update && apt-get install -y netcat curl dnsutils net-tools vim && apt-get clean
+RUN apt-get update && apt-get install -y netcat curl dnsutils net-tools postgresql-client && apt-get clean
```
Comment (mhoppa) [inline infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile]: i wonder if someone had vim installed for debugging purposes?

Comment (ocket8888) [reply infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile]: Yeah, that's how it usually happens

**Pull request merged**